src/main/resources/org/jenkinsci/plugins/docker/workflow/Docker.groovy

```diff
@@ -137,7 +137,8 @@ class Docker implements Serializable {
 
         public Container run(String args = '', String command = "") {
             docker.node {
-                def container = docker.script."${docker.shell()}"(script: "docker run -d${args != '' ? ' ' + args : ''} ${id}${command != '' ? ' ' + command : ''}", returnStdout: true).trim()
+                def shouldUseDetach = !args.contains(' -a') && !args.contains("--attach")
+                def container = docker.script."${docker.shell()}"(script: "docker run ${shouldUseDetach? -d : ''}${args != '' ? ' ' + args : ''} ${id}${command != '' ? ' ' + command : ''}", returnStdout: true).trim()
```

Comment:
So we're passing the `-a` or `--attach` to `docker run` - is that the intent? (I'm just trying to understand the usage case). Could you provide and example of how you'd call this?